spr/src/commands/init.rs

```diff
@@ -184,8 +185,112 @@ pub async fn init() -> Result<()> {
     let branch_prefix = dialoguer::Input::<String>::new()
         .with_prompt("Branch prefix")
         .with_initial_text(branch_prefix)
+        .validate_with(|input: &String| -> Result<()> {
+            validate_branch_prefix(input)
+        })
         .interact_text()?;
+
     config.set_str("spr.branchPrefix", &branch_prefix)?;
 
     Ok(())
 }
+
+fn validate_branch_prefix(branch_prefix: &str) -> Result<()> {
+    // They can include slash / for hierarchical (directory) grouping, but no slash-separated component can begin with a dot . or end with the sequence .lock.
+    if branch_prefix.contains("/.")
```

Comment:
Do you need to check `branch_prefix.starts_with(".")`, too?

Reply:
Good point. I missed the fact that due to the prefix, we get a "/" at the beginning anyway and thus it's not enough to just check for `"/.`.

I amended the commit.